### PR TITLE
Adopt new way of getting notification data in PushNotificationBackgroundSynchronizer

### LINKED ISSUE #13852: Adopt new way of getting notification data in PushNotificationBackgroundSynchronizer
More context in https://github.com/woocommerce/woocommerce-ios/pull/13734#pullrequestreview-2276022786

### PR DESCRIPTION
**Pull request by bozidarsevo**
Closes: #13852

## Description

Since we might already have `note` in the push notification (parsed from `note_full_data` part of the payload) we can check if we already have `orderID` in it without the need of syncronizing all the notifications and avoid extra network calls.
If we have the `orderID` we can go directly to syncronizing that **exact** order.

In this PR we use that ability and we use `orderID` from the `note` if we have that property set/loaded.

## Testing information

Since this PR changes are not visible from the UI, testing can be done by adding some breakpoints in the `PushNotificationBackgroundSynchronizer`:
- add breakpoint to line 40 -> `orderID = Int64(noteOrderID)`
- run the app from Xcode
- put the app in the background
- get a new order notification
- check that `sync()` function in `PushNotificationBackgroundSynchronizer` is called and debugger stops on that breakpoint at line 40
- check that the notification has the proper `orderID

Screenshot 2024-09-12 at 11 39 12

Main breakpoint:

Screenshot 2024-09-12 at 11 39 17

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

File: WooCommerce/Classes/Notifications/PushNotificationBackgroundSynchronizer.swift [hunk 1]
```diff
@@ -27,22 +27,27 @@ struct PushNotificationBackgroundSynchronizer {
...
-
-            // Find the orderID from the previously synced notification.
-            guard let orderID = getOrderID(noteID: pushNotification.noteID) else {
-                return .newData
+            let orderID: Int64
+
+            if let note = pushNotification.note,
+               let noteOrderID = note.meta.identifier(forKey: .order) {
+                orderID = Int64(noteOrderID)
+            } else {
```
Comment (bozidarsevo) [inline WooCommerce/Classes/Notifications/PushNotificationBackgroundSynchronizer.swift:41]: This the only change, checking if we already have the orderID in the notification `note` property.

**Review by Ecarrion — COMMENTED** [review]
Good job @bozidarsevo, code looks good and tests well!

However, before approving, I think we can add a unit test on the `PushNotificationBackgroundSynchronizerTests` file.

**Comment by bozidarsevo** [comment]
> However, before approving, I think we can add a unit test on the `PushNotificationBackgroundSynchronizerTests` file.

What do you think, what kind of test case would be ok?
Maybe to check that `synchronizeNotifications()` was not called if the push notification has the orderID, or some other test?

**Comment by Ecarrion** [comment]
> > However, before approving, I think we can add a unit test on the `PushNotificationBackgroundSynchronizerTests` file.
> 
> What do you think, what kind of test case would be ok? Maybe to check that `synchronizeNotifications()` was not called if the push notification has the orderID, or some other test?

I was thinking on something like:

```
func test_synchronizer_stores_order_notification_when_order_comes_inside_notification() {
    
    // Given
    mock_network_requests_without_notifications_endpoint()
    
    // When
    initialize_synchronizer_with_full_notification_data()
    
    // Then
    check_that_order_is_correctly_stored()
}
```

Does that make sense?

**Review by Ecarrion — APPROVED** [review]

**Pull request merged**